### ISSUE TITLE
[FEATURE] Importing API -> match entries in array by custom 'idField'

### ISSUE THREAD

**Issue by deploy-aurocraft** [Baboo7/strapi-plugin-import-export-entries#53]
**Is your feature request related to a problem? Please describe.**

- For now by using Import API we're able to choose `idField` attribute to change identifier by which main entries can be updated or created.
- IF we make an import of list of Entries which have dependencies with other Collection Types - matching with those entries is only possible if `id` field is ised in relation array or object
- But if it would be possible to configure `idField` also for related entries (for example by `slug`) - it may scale possibility to make import of multidependencies.

**Case example**
```
{
    "slug": "api::product.product",
    "format": "json",
    "data":[
                { 
                "name": "nameTest", 
                "crmId": "CrmIdTest", 
                "unit": {"id":1, "slug": "string"}
                }
            ],
    "idField": "crmId"
}
```

in case if `unit.id` is included in import request it will match with already created **Unit** entry, but if id is not included -> the request will create new **Unit** entry

**Possible solution options**
1) If the default id configuration field may be configured through `plugins.js` file for all relations

2) Possibility to insert `idField` inside relation object or array

Example:
```
{
    "slug": "api::product.product",
    "format": "json",
    "data":[
                { 
                "name": "nameTest", 
                "crmId": "CrmIdTest", 
                "unit": [data: {"id":1, "slug": "string", }, "idField": "crmId"]
                }
            ],
    "idField": "crmId"
}
```

or any other solution will be wonderful

Thank you for any feedback!

**Comment by Alex01d** [comment]
I think of a hierarchy of identifier fields, like id, then slug, then title or something like that.

**Comment by deploy-aurocraft** [comment]
The only possible field in strapi is `id` so all others should be configured

**Comment by Alex01d** [comment]
Testing this one: https://github.com/Alex01d/strapi-plugin-import-export-entries/tree/idField-hierarchy

Config variable example: importUniqueIdentifierField: ['id', 'slug', 'url', 'title', 'name', 'legend']

**Comment by deploy-aurocraft** [comment]
you're superhero

**Comment by yuriyglushakov** [comment]
@Alex01d Should it be configured via plugins.js in this case?

**Comment by Alex01d** [comment]
> @Alex01d Should it be configured via plugins.js in this case?

Yes:
`module.exports = ({ env }) => ({
  'import-export-entries': {
    enabled: true,
    config: {
      importUniqueIdentifierField: ['id', 'slug', 'url', 'title', 'name', 'legend'],
    },
  },
});`

**Comment by yuriyglushakov** [comment]
Working perfectly, thank you! Was testing with obj fields (will check on arrays bit later)

**Comment by Baboo7** [comment]
Hey everyone, FYI I just published version `1.15.0` where v2 of export/import has been implemented. It's a beta still but it showed more stable results than the v1 on the test data I used. Export v2 is accessible in the export modal by selecting the export format `JSON (v2)` / import v2 is detected automatically from the structure of the provided data file. This new version `1.15.0` does NOT introduce breaking changes so v1 is still accessible and behaving as in previous version 😉 

With this new file structure, it will be easier and more maintainable to have custom id fields per collection types. Since this feature is in high demand, it will be prioritized in a near future.

I would love to have your feedback on this new version. Feel free to join the [discord server](https://discord.gg/cGRJh7M7) of the project I just opened 😌

**Comment by yuriyglushakov** [comment]
Thanks for this update @Baboo7! Will test on weekends! Discord invite link seems not working :( Could you please check it?

**Comment by Baboo7** [comment]
Whoo thanks for the feedback here is the new link https://discord.gg/dcqCAFFdP8

**Comment by Baboo7** [comment]
This issue is solved in version 1.19.0. it is now possible to define an idField per collection. Check out the documentation at the section Config
